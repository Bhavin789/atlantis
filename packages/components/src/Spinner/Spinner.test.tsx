import React from "react";
import { render } from "@testing-library/react";
import { Spinner } from ".";

it("renders the spinner", () => {
  const { container } = render(<Spinner />);
  expect(container).toMatchSnapshot();
});

it("renders the spinner with content", () => {
  const { container, getByRole } = render(<Spinner>Content</Spinner>);
  expect(container).toMatchSnapshot();
  expect(getByRole("alert")).toBeInTheDocument();
});

it("doesnt render the spinner when not loading", () => {
  const { container, queryByRole } = render(
    <Spinner loading={false}>Content</Spinner>,
  );
  expect(container).toMatchSnapshot();
  expect(queryByRole("alert")).not.toBeInTheDocument();
});

it("renders the small spinner", () => {
  const { container } = render(<Spinner size="small" />);
  expect(container).toMatchSnapshot();
});

it("renders the inline spinner", () => {
  const { container } = render(<Spinner inline={true} />);
  expect(container).toMatchSnapshot();
});

it("renders a small inline spinner", () => {
  const { container } = render(<Spinner inline={true} size="small" />);
  expect(container).toMatchSnapshot();
});
