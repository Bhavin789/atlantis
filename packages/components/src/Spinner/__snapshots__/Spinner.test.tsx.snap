// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesnt render the spinner when not loading 1`] = `
<div>
  Content
</div>
`;

exports[`renders a small inline spinner 1`] = `
<div>
  <div
    class="wrapper inline"
  >
    <div
      aria-busy="true"
      aria-hidden="true"
      aria-live="polite"
    />
    <div
      class=""
    >
      <div
        aria-busy="true"
        aria-label="loading"
        class="spinner small inline"
        role="alert"
      />
    </div>
  </div>
</div>
`;

exports[`renders the inline spinner 1`] = `
<div>
  <div
    class="wrapper inline"
  >
    <div
      aria-busy="true"
      aria-hidden="true"
      aria-live="polite"
    />
    <div
      class=""
    >
      <div
        aria-busy="true"
        aria-label="loading"
        class="spinner inline"
        role="alert"
      />
    </div>
  </div>
</div>
`;

exports[`renders the small spinner 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      aria-busy="true"
      aria-hidden="true"
      aria-live="polite"
    />
    <div
      class="overlay"
    >
      <div
        aria-busy="true"
        aria-label="loading"
        class="spinner small"
        role="alert"
      />
    </div>
  </div>
</div>
`;

exports[`renders the spinner 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      aria-busy="true"
      aria-hidden="true"
      aria-live="polite"
    />
    <div
      class="overlay"
    >
      <div
        aria-busy="true"
        aria-label="loading"
        class="spinner"
        role="alert"
      />
    </div>
  </div>
</div>
`;

exports[`renders the spinner with content 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      aria-busy="true"
      aria-hidden="true"
      aria-live="polite"
    >
      Content
    </div>
    <div
      class="overlay"
    >
      <div
        aria-busy="true"
        aria-label="loading"
        class="spinner"
        role="alert"
      />
    </div>
  </div>
</div>
`;
